node/declaration.ts

```diff
@@ -44,13 +44,13 @@ export const getDeclarationsFromConfig = (
       const paths = Array.isArray(config.path) ? config.path : [config.path]
 
       paths.forEach((path) => {
-        declarations.push({ ...declaration, ...config, path })
+        declarations.push({ ...declaration, cache: config.cache, path })
```

Comment:
Not sure I understand this bit. Why are we suddenly setting the `cache` property specifically rather than all of `config`? And does it mean that if we add more configuration properties to ISC, will we have to add them here in addition to `config.ts`?

Reply:
This is necessary because `config` now not only contains `cache` and `path`, but also `excludedPatterns`, which should not be part of the `declaration`. This means that we'll have to manually add all declaration-specific ISC properties here, yes.

Now that I think about it, should `preCache` / `postCache` be part of functions config as well? Are there cases where the same edge function sometimes should run before, and sometimes after cache? (this is out of scope for this PR, but nevertheless we should think about it)

Reply:
> This is necessary because `config` now not only contains `cache` and `path`, but also `excludedPatterns`, which should not be part of the `declaration`.

I guess you could push to the declaration everything except `excludedPatterns`, which you know shouldn't be there. If we add more properties to ISC we wouldn't need to manually include them here?

Happy to defer to you though.